test/models/user_test.rb

```diff
@@ -0,0 +1,64 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+
+  test "validations: username can't be blank" do
+    user = User.new
+
+    assert_not user.valid?
+    assert user.errors.keys.include?(:username), "username is not in the errors hash"
```

Comment:
does this show up as an error on the User Interface?  If so, it could be written to be readable to a general user.